apis/management.cattle.io/v3/rke_types.go

```diff
@@ -400,6 +400,12 @@ type IngressConfig struct {
 	ExtraArgs map[string]string `yaml:"extra_args" json:"extraArgs,omitempty"`
 	// DNS Policy
 	DNSPolicy string `yaml:"dns_policy" json:"dnsPolicy,omitempty"`
+	// Extra Env vars
+	ExtraEnvs []v1.EnvVar `yaml:"extra_envs" json:"extraEnvs,omitempty" norman:"type=array[json]"`
+	// Extra volumes
+	ExtraVolumes []v1.Volume `yaml:"extra_volumes" json:"extraVolumes,omitempty" norman:"type=array[json]"`
```

Comment:
@mrajashree can you explore if it's possible to expose the field in our api as a concrete type []v1.Volume vs array[json]

Reply:
As per offline discussion, sticking to array[json] so that the "Edit as YAML" does not add unset fields with default nil values.
But we'll use embedded types so that when we import them in schema.go we get a v3/schema type for each, so users can refer that for input.
And there is custom api level schema validation check on rancher side for these fields